scripts/oderIdValidation.js

```diff
@@ -0,0 +1,78 @@
+/**
+ * We import these just for type checking as the actual imports
+ * are passed as arguments to the function
+ * https://github.com/marketplace/actions/github-script#run-a-separate-file
+ */
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const core = require('@actions/core');
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+const github = require('@actions/github');
+
+/**
+ *
+ * @param {core} core
+ * @param {github} github
+ */
+// eslint-disable-next-line @typescript-eslint/no-shadow
+module.exports = async (core, github) => {
+  try {
+    const owner = process.env.OWNER;
+    const repo = process.env.REPO;
+    const issueNumber = process.env.ISSUE_NUMBER;
+    const token = process.env.TOKEN;
+
+    const orderApiToken = process.env.ORDER_API_TOKEN;
+    const orderApi = 'https://store-wp.mui.com/wp-json/wc/v3/orders/';
+
+    const octokit = github.getOctokit(token);
+
+    const issue = octokit.rest.issues.get({
+      owner,
+      repo,
+      issue_number: issueNumber,
+    });
+
+    // add to this regex the possibility that the ORDER ID is wrapped in ** or __
+    const orderIdRegex = /(?:\*|_){0,2}?Order ID(?:\*|_){0,2}?: (\d+)/;
+    const orderIdMatch = issue.data.body.match(orderIdRegex)[1];
+    const orderId = orderIdMatch ? orderIdMatch[1] : null;
+
+    if (!orderId) {
+      core.setFailed('No Order ID found in issue body');
+    } else {
+      const order = await fetch(`${orderApi}${orderId}`, {
+        headers: {
+          Authorization: orderApiToken,
+          'User-Agent': 'MUI-Tools-Private/X-Orders-Inspector v1',
+        },
+      });
+      const orderDetails = await order.json();
+      const plan =
+        orderDetails.line_items?.filter((item) => item.name.test(/\b(pro|premium)\b/i))[0].name ||
```

Comment:
We have products that have "pro" in their title, e.g. https://mui.com/store/items/devias-kit-pro/. This feels a bit too brittle. Instead, we can get inspiration from the keymailer: https://github.com/mui/mui-private/blob/09f2c3f4263db001a442c6f3b51cb32f935a2e20/lambda/keymailer/index.js#L149.

Reply:
Actually @hasdfa I could see us publishing https://github.com/mui/mui-private/blob/09f2c3f4263db001a442c6f3b51cb32f935a2e20/lambda/keymailer/databaseMuiX.js#L20 as an npm package, and then use it in the GitHub Action, Pipedream, Toolpad App. Or to be fair, a REST API might be as clean.